hw4.Rmd

```diff
@@ -30,43 +33,84 @@ Before doing anything else, just play with the data to get a sense of it.
+  filter(decision == "yes") %>%
+  summarise(yes = n()) 
+
+cases$yes <- rate_yes$yes
+
+cases <- cases %>%
+  mutate(rate_yes = (yes/cases))
+
+p1 <- ggplot(data = cases, aes(x = judge, y = rate_yes)) + 
+  geom_bar(stat = "identity") +
+  labs(x = "Judge", y = "Rate of Granting Appeals") +
+  ggtitle("Rate of Granting Appeals per Judge")
+p1
+
 ```
 
-(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
+(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whether an appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+data$judge <- relevel(as.factor(data$judge), ref="Iacobucci")
+fit <- glm(decision ~ judge, family = "binomial", data = data)
+tidy(fit)
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+b <- coef(fit)
+b.marceau <- b["judgeMarceau"]
+exp(b.marceau)
+plogis(b[1] + b.marceau)
 ```
+The predicted probability that Judge Marceau will grant an appeal is 60%.
 
 (@) Perhaps the different judges have different types of cases.  That might be why they seem to be making such different decisions.  Fit a model treating judge as a factor plus the following predictors: (i) an indicator for whether the expert rater thought the case had merit, (ii) location indicators using 'other' as the reference category, (iii) a language dummy with French as omitted category.  Note that as above you might have to relevel the factors to get this to work as you want.
 
 ```{r}
-# [your code here]
+data$location <- relevel(as.factor(data$location), ref = "other")
+data$language <- relevel(as.factor(data$language), ref = "French")
+fit2 <- glm(decision ~ judge + rater + location + language, data = data, family = binomial)
+tidy(fit2)
 ```
 
 (@) For a case that was ruled to have merit, that took place in Toronto, and that was in English, what is the predicted probability that Judge Marceau will grant the appeal?
 
 ```{r}
-# [your code here]
+# I need more help to fully understand this one. 
+#data.mean <- with(fit2, data.frame(rater = yes, location = Toronto, language = English))
```

Comment:
You could also just do this "by hand" using the equation we discussed in class and lab. See solutions and the slides for a refresher.